jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -52,10 +63,21 @@
    * @param sourceFile the source file to add to the layer
    * @param extractionPath the path in the container file system corresponding to the {@code
    *     sourceFile}
+   * @param permissions the file permissions on the container. Use {@code null} to use defaults (644
+   *     for files, 755 for directories)
    */
-  public LayerEntry(Path sourceFile, AbsoluteUnixPath extractionPath) {
+  public LayerEntry(
+      Path sourceFile,
+      AbsoluteUnixPath extractionPath,
+      @Nullable Set<PosixFilePermission> permissions) {
```

Comment:
Given all the changes to add a `null` parameter, it might be best to retain the original 2-argument contractor.

Reply:
Almost all of the changes were in tests, and `LayerEntry` isn't supposed to be used by jib-core users, so I think it's fine.